### PR TITLE
Update WPF to .NET 5

### PR DESCRIPTION
**Pull request by brianlagunas**
﻿## Description of Change

Updated Prism for WPF to .NET 5

### PR Checklist

- [x] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

File: src/Prism.Core/Prism.Core.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netstandard2.0;net45;net47</TargetFrameworks>
+    <TargetFrameworks>netstandard2.1;net45;net47;net5.0</TargetFrameworks>
```
Comment (dansiegel) [inline src/Prism.Core/Prism.Core.csproj:4]: please revert the netstandard target to 2.0. There are too many consumers out there using netstandard2.0. We need to continue supporting netstandard2.0 for the foreseeable future.

**Review by dansiegel — CHANGES_REQUESTED** [review]

File: tests/Wpf/Prism.DryIoc.Wpf.Tests/Prism.DryIoc.Wpf.Tests.csproj [hunk 2]
```diff
@@ -2,7 +2,7 @@
 <Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">
 
   <PropertyGroup>
-    <TargetFramework>net461</TargetFramework>
+    <TargetFramework>net472</TargetFramework>
```
Comment (dansiegel) [inline tests/Wpf/Prism.DryIoc.Wpf.Tests/Prism.DryIoc.Wpf.Tests.csproj:5]: don't you want to update this like you did the Prism.Core.Tests to multi-target?

File: tests/Wpf/Prism.Unity.Wpf.Tests/Prism.Unity.Wpf.Tests.csproj [hunk 3]
```diff
@@ -2,7 +2,7 @@
 <Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">
 
   <PropertyGroup>
-    <TargetFramework>net461</TargetFramework>
+    <TargetFramework>net472</TargetFramework>
```
Comment (dansiegel) [inline tests/Wpf/Prism.Unity.Wpf.Tests/Prism.Unity.Wpf.Tests.csproj:5]: don't you want to update this like you did the Prism.Core.Tests to multi-target?

File: tests/Wpf/Prism.Wpf.Tests/Prism.Wpf.Tests.csproj [hunk 4]
```diff
@@ -1,7 +1,7 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">
   <PropertyGroup>
-    <TargetFramework>net461</TargetFramework>
+    <TargetFramework>net472</TargetFramework>
```
Comment (dansiegel) [inline tests/Wpf/Prism.Wpf.Tests/Prism.Wpf.Tests.csproj:4]: don't you want to update this like you did the Prism.Core.Tests to multi-target?

**Review by dansiegel — CHANGES_REQUESTED** [review]

File: tests/Wpf/Prism.DryIoc.Wpf.Tests/Prism.DryIoc.Wpf.Tests.csproj (hunk 2 above)
Comment (brianlagunas) [reply tests/Wpf/Prism.DryIoc.Wpf.Tests/Prism.DryIoc.Wpf.Tests.csproj:5]: not in this PR, that will be a separate effort

File: tests/Wpf/Prism.Unity.Wpf.Tests/Prism.Unity.Wpf.Tests.csproj (hunk 3 above)
Comment (brianlagunas) [reply tests/Wpf/Prism.Unity.Wpf.Tests/Prism.Unity.Wpf.Tests.csproj:5]: not in this PR, that will be a separate effort

File: tests/Wpf/Prism.Wpf.Tests/Prism.Wpf.Tests.csproj (hunk 4 above)
Comment (brianlagunas) [reply tests/Wpf/Prism.Wpf.Tests/Prism.Wpf.Tests.csproj:4]: not in this PR, that will be a separate effort

**Comment by kekonn** [comment]
What else do you need to make this happen? I'm waiting on this to become real to start a project.

**Comment by brianlagunas** [comment]
@kekonn a passing build. Apparently, Azure Devops hasn't been updated to work with .NET 5 yet.

**Comment by kekonn** [comment]
That's.... Rough

**Comment by kekonn** [comment]
@brianlagunas I'm not sure about Azure, but Github Action's default .net core runner supports .NET 5. I've already used it to build .NET 5 projects. Is it a big conversion?

**Comment by brianlagunas** [comment]
Yay! Can't wait for that.

File: build/steps/prepare-build.yml [hunk 5]
```diff
@@ -3,7 +3,13 @@ parameters:
 
 steps:
 - task: UseDotNet@2
-  displayName: Use .NET 3.1
+  displayName: Ensure 3.1 SDK is Installed
+  inputs:
+    packageType: 'sdk'
+    version: '3.1.404'
```
Comment (calebnelton) [inline build/steps/prepare-build.yml:9]: Should this be changed to '5.0.x'

It looks like Azure DevOpe Pipelines does support .net5.0 apps now:
[https://stackoverflow.com/questions/64830501/building-net-5-0-project-azure-devops-pipeline](https://stackoverflow.com/questions/64830501/building-net-5-0-project-azure-devops-pipeline)

Comment (dansiegel) [reply build/steps/prepare-build.yml:9]: @calebnelton you may notice that we use a global.json, the hardcoded 3.1 sdk is a hack for Azure DevOps

**Review by dansiegel — APPROVED** [review]

**Pull request merged**